File: admin/web/src/api/ddz/arenaPeriod.js
import service from '@/utils/request'

// 获取期号列表
export const getArenaPeriodList = (data) => {
  return service({
    url: '/ddz/arenaPeriod/list',
    method: 'post',
    data
  })
}

// 获取期号详情
export const getArenaPeriodInfo = (id) => {
  return service({
    url: '/ddz/arenaPeriod/info',
    method: 'get',
    params: { id }
  })
}

// 更新期号
export const updateArenaPeriod = (data) => {
  return service({
    url: '/ddz/arenaPeriod/update',
    method: 'put',
    data
  })
}

// 删除期号
export const deleteArenaPeriod = (data) => {
  return service({
    url: '/ddz/arenaPeriod/delete',
    method: 'delete',
    data
  })
}

// 获取期号玩家列表
export const getArenaPeriodPlayers = (data) => {
  return service({
    url: '/ddz/arenaPeriod/players',
    method: 'post',
    data
  })
}

// 获取期号报名日志
export const getArenaPeriodSignupLogs = (data) => {
  return service({
    url: '/ddz/arenaPeriod/signupLogs',
    method: 'post',
    data
  })
}

// 获取期号统计
export const getArenaPeriodStats = () => {
  return service({
    url: '/ddz/arenaPeriod/stats',
    method: 'get'
  })
}

File: admin/server/router/ddz/arena_period.go
package ddz

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DDZArenaPeriodRouter struct{}

func (r *DDZArenaPeriodRouter) InitDDZArenaPeriodRouter(Router *gin.RouterGroup) {
	ddzArenaPeriodRouter := Router.Group("ddz/arenaPeriod").Use(middleware.OperationRecord())
	ddzArenaPeriodRouterWithoutRecord := Router.Group("ddz/arenaPeriod")

	ddzArenaPeriodApi := v1.ApiGroupApp.DDZApiGroup.DDZArenaPeriodApi
	{
		ddzArenaPeriodRouter.PUT("update", ddzArenaPeriodApi.UpdateArenaPeriod)     // 更新期号
		ddzArenaPeriodRouter.DELETE("delete", ddzArenaPeriodApi.DeleteArenaPeriod) // 删除期号
	}
	{
		ddzArenaPeriodRouterWithoutRecord.POST("list", ddzArenaPeriodApi.GetArenaPeriodList)         // 获取期号列表
		ddzArenaPeriodRouterWithoutRecord.GET("info", ddzArenaPeriodApi.GetArenaPeriodByID)          // 获取期号详情
		ddzArenaPeriodRouterWithoutRecord.POST("players", ddzArenaPeriodApi.GetArenaPeriodPlayers)   // 获取期号玩家列表
		ddzArenaPeriodRouterWithoutRecord.POST("signupLogs", ddzArenaPeriodApi.GetArenaPeriodSignupLogs) // 获取期号报名日志
		ddzArenaPeriodRouterWithoutRecord.GET("stats", ddzArenaPeriodApi.GetArenaPeriodStats)        // 获取期号统计
	}
}
